### PR TITLE
Addressing MX notes per spec 1.35

### PR DESCRIPTION
**Pull request by amartinez-cg**
minor tidying of header files, moved function open bracket to own line.  Removes some whitespace.

mx_scale: add center back to result in vscale
mx_luminance dot prods against luma coeffs
mx_hueshift amount defaults to 0.5
mx_dot for vector2 and vector4 implemented per spec

## Checklist:

- [ ] I have read the [contribution guidelines](../CONTRIBUTING.md).
- [ ] I have previously submitted a [Contributor License Agreement](http://opensource.imageworks.com/cla/).
- [ ] I have updated the documentation, if applicable.
- [ ] I have ensured that the change is tested somewhere in the testsuite (adding new test cases if necessary).
- [ ] My code follows the prevailing code style of this project.

### CODE REVIEW

File: src/shaders/MaterialX/vector4.h [hunk 1]
```diff
@@ -338,7 +369,8 @@ vector4 min(vector4 in, float amount){
     return out;
 }
 
-vector4 fmod(vector4 in, vector4 amount){
+vector4 fmod(vector4 in, vector4 amount)
+{
     vector4 out = in;
```
Comment (lgritz) [inline src/shaders/MaterialX/vector4.h:374]: I see this idiom all over the place.

    T out = in;
    T out.x = ... // reassign
    T out.y = ... // reassign
    ...
    return out;

Why bother with the out=in if you're just going to write over it?

And for that matter, you don't need out at all, you can use constructors:

    return vector4 (fmod (in.x, amount.x),
                    fmod (in.y, amount.y),
                    fmod (in.z, amount.z),
                    fmod (in.w, amount.w),
);

I realize that what I'm suggesting is a pervasive and intrusive change. I am not asking that you do this as part of this checkin. Just something to think about for later.

File: src/shaders/MaterialX/funcs.h [hunk 2]
```diff
@@ -81,34 +83,52 @@ color fgamma(color in, float g){ return sign(in) * pow(abs(in), g);}
...
     out *= amount;
     out += pivot;
     return out;
 }
-float exponent(float in, float amount){ return (in > 0) ? pow(in, amount) : in;}
-color exponent(color in, color amount){ return (luminance(in) > 0) ? pow(in, amount) : in; }
-color exponent(color in, float amount){ return (luminance(in) > 0) ? pow(in, amount) : in; }
+float exponent(float in, float amount)
+{ 
+    return (in > 0) ? pow(in, amount) : in;
```
Comment (lgritz) [inline src/shaders/MaterialX/funcs.h:111]: I think exponent(in,amount) is just OSL's pow(in,amount). It's already a "safe" version.

File: src/shaders/MaterialX/funcs.h [hunk 3]
```diff
@@ -81,34 +83,52 @@ color fgamma(color in, float g){ return sign(in) * pow(abs(in), g);}
...
+}
+color exponent(color in, color amount)
+{ 
+    return (luminance(in) > 0) ? pow(in, amount) : in;
```
Comment (lgritz) [inline src/shaders/MaterialX/funcs.h:115]: Doug is correct, this doesn't make sense. luminance(in) > 0 does not guarantee that all of the individual components are > 0.

**Comment by lgritz** [comment]
Tip for next time: it's very hard to review a small number of logic changes amid a sea of formatting changes. I would recommend doing any widespread formatting changes as an isolated check, which you can say explicitly is a formatting-only change, with no need to review line-by-line if a representative glance at the new style looks acceptable. Then any changes to actual shader behavior or logic can be examined without being obfuscated by unrelated style or formatting changes.

**Comment by lgritz** [comment]
But anyway, LGTM!

When you're ready to merge, click the little down arrow on the "merge" button and select "squash and merge". Then click the resulting button. That will avoid a redundant merge commit just for this one change.

**Comment by amartinez-cg** [comment]
I will make these changes and resubmit.  At this point nothing is too
pervasive and intrusive.  We've written this library twice already.

**Pull request merged**